pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -559,6 +597,10 @@ func (host *BareMetalHost) HardwareProfile() string {
 // SetHardwareProfile updates the hardware profile name and returns
 // true when a change is made or false when no change is made.
 func (host *BareMetalHost) SetHardwareProfile(name string) (dirty bool) {
+	if host.Status.HardwareProfile != name {
```

Comment:
why is that for? Isn't the same as lower code?

Reply:
Removed